### DISCUSSION TITLE
Startup Script

### DISCUSSION THREAD

**Discussion by bkbilly** [TECH7Fox/asterisk-hass-addons#336] [Ideas]
It would be nice to run a script when the Addon starts which could be used for a lot of reasons like:
 1. Downloading code or files from the internet
 2. Installing Python dependencies for using in AGI scripts
 3. Insert records to asterisk database

Ideally this could be done with a text input at the configuration that could create a bash script and runs when it boots.

Also I noticed that it has access to the `/media` folder, but the access to the `/share` folder is missing.

**Comment by bkbilly** [comment]
I've looked into the available options, and the text input is not supported, so it could be a list of commands of type str.

**Comment by TECH7Fox** [comment]
I like this idea. I'm also looking to install a python script in Asterisk and to accept a script to run on startup would be nice wide solution.

So you could place a `startup.sh` script in the config folder, and it would check if that exists and run it if it does. Then you can do whatever you want there.

**Reply by TECH7Fox** [reply]
I will play around with it this week.

**Reply by bkbilly** [reply]
A `startup.sh` script is a nice idea to have.
The best scenario would be to start after the asterisk CLI starts so that it can run Asterisk CLI commands.
I am not sure how this check could be done, but it would be very helpful.

**Reply by TECH7Fox** [reply]
I added this feature in v4.4.0. If a `/config/asterisk/startup.sh` exists the add-on will run it at startup. It doesn't wait for Asterisk to start as that would bring a lot of extra complexity and other issues. But you should be able to do everything you want in the script, or if you really want to do something in Asterisk at runtime you could try sending commands to the AMI. (maybe using the Asterisk integration)

**Reply by bkbilly** [reply]
That's awesome! Thank you for this, it will be very helpful!

**Comment by bkbilly** [comment]
It would be nice for the docker image to come with the `ffmpeg` package because installing it using the startup script takes several minutes to complete.
This is a requirement of the `pydub` python library which I am using for my AGI scripts.
